### PR TITLE
IC-1286: Appointment details form

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
## What does this pull request do?

This displays a form which allows a service provider user to see the details of an appointment. It does not provide the ability to actually update the appointment, which will come next.

## What is the intent behind these changes?

To continue building the service user journey.

## Screenshot

Screenshot_2021-03-29 HMPPS Interventions - GOV UK

### CODE REVIEW

**Review by Gweaton — COMMENTED** [review]
I'm finding this a bit difficult to review all in one go, would it be a real pain to split this out into a couple of smaller PRs? e.g.
- One for refactoring dates/times
- One for adding the script to generate types for custom component macro args
- One for adding the new form

File: server/utils/time.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+export default class Time {
+  /*
+   * 0 ≤ hour ≤ 23
+   * 0 ≤ minute ≤ 59
+   * 0 ≤ second ≤ 59
+   */
+  private constructor(readonly hour: number, readonly minute: number, readonly second: number) {}
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply server/utils/time.ts]: We talked about this one offline - I explained that this class is meant to provide a timezone-independent way of representing the values displayed on a clock.

**Comment by lawrence-forooghian** [comment]
@Gweaton thanks for suggesting that I split this out into separate PRs - that's got this one down to a much more manageable 2 commits! It's ready for re-review now.

**Review by Gweaton — APPROVED** [review]
Looks good, a lot easier to review!

**Pull request merged**